clients/stellar-relay-lib/src/connection/connector/message_reader.rs

```diff
@@ -141,17 +116,14 @@ async fn read_message_from_stellar(stream_clone: Arc<Mutex<TcpStream>>) -> Resul
 				// The next few bytes was read. Add it to the readbuf.
 				lack_bytes_from_prev -= size;
 				readbuf.append(&mut buff_for_reading);
+				// make sure to cleanup the buffer
+				buff_for_reading = vec![0; 4];
```

Comment:
Why did we not needed before to clean the buffer at this stage? I imagined that if we don' clean it, on the next read it would be overwritten. Or is it because it may read less than 4 and therefore have partially old data on it?
Anyway it seems a good addition!

Reply:
This is actually a surprise I found: 
Some of the integ tests failed because a proof could not be formed. I logged [`fn listen()` of overlay](https://github.com/pendulum-chain/spacewalk/blob/cce0b3d25bf9e4b8add7b53fe38f1cdad1f74791/clients/stellar-relay-lib/src/overlay.rs#L75) and it was always empty, even after sending a message to Stellar Node that would send back a legit Stellar Message.

It seems like the bytes concatenated was wrong; I had to flush the first few bytes before reading a next batch.

The first 4 bytes is to determine the length (how many bytes) of the next Stellar Message .

Reply:
Makes sense!